### PR TITLE
Improved code on Posting,EditPostForm and Search Component

### PR DESCRIPTION
**Pull request by niniwoo**
-Display username instead of user id on Posting component
Screenshot 2024-01-05 at 8 32 46 PM

-Display username instead of user id in Post Detail 
-Display category name instead of category id in Post Detail
Screenshot 2024-01-05 at 8 33 02 PM

-Fetch category data on EditPostForm
Screenshot 2024-01-05 at 8 33 14 PM

-Enable to search posting by username
Screenshot 2024-01-05 at 8 34 15 PM

### CODE REVIEW

File: frontend/src/components/Posting.jsx
Comment (sulgikim) [inline frontend/src/components/Posting.jsx:1]: Just a quick note, the name of component could be PostsList and the PostingDetail could be PostDetail. "Posting" only was a bit confusing name to me.

**Review by sulgikim — APPROVED** [review]
Looks good to me :)

File: backend/controllers/postController.js [hunk 1]
```diff
@@ -87,11 +87,12 @@ const createPost = asyncHandler(async (req, res) => {
 const updatePost = asyncHandler(async (req, res) => {
   const { title, imageUrl, category, content, postVisibility } = req.body;
   const post = await Post.findById(req.params.id);
-
+  const categoryID = await Category.findById(category);
```
Comment (hendrik-kim) [inline backend/controllers/postController.js:90]: I think Category is a better variable name than CategoryID. It's an object that contains an ID and other info data.

File: backend/models/postModel.js [hunk 2]
```diff
@@ -11,6 +11,10 @@ const postSchema = mongoose.Schema({
     required: true,
     ref: "User",
   },
+  username: {
```
Comment (hendrik-kim) [inline backend/models/postModel.js:14]: It's questionable that if it's appropriate to have the usename property in the Post Model itself, which should be the minimum unit for that entity.  Wouldn't it be more appropriate to just hold the user ID and then find and include the username in the business logic if needed?

**Review by hendrik-kim — APPROVED** [review]

**Pull request merged**